textBasedAdventure.js

```diff
@@ -1,5 +1,583 @@
+    case "oatmeal":
+      console.clear();
+      oatmeal(); 
+      break;
+    case "sandwiches":
+        console.clear();
+      sandwiches();
+      break;
+    default:
+        console.clear();
+        smoothieBowls();
+  }
+}
+      
+let wantBreakfast = null;
+while (wantBreakfast !== "y") {
+  wantBreakfast = readline.question('Would you like to prepare a nice BREAKFAST to start your day, [y]es or [n]o? ');
+  // console.log(wantBreakfast)
+}
+      
+  console.log("Let see what you have first as options.");
+
+let willUserChoose = readline.question("would you like to choose for your[self] or, be given a [random] option? ");
+//  for (let i = 0; i < recipes.length; i++) {
+  function getRecipe() {
+      
+    let recipes = ["oatmeal", "sandwiches", "smoothie bowls"];
+    let selectedRecipe = recipes[Math.floor(Math.random() * recipes.length)];
+    return selectedRecipe;
+
+  }
+  
+  if (willUserChoose === "random") {
+  
+      let selected = getRecipe();
+      console.log(selected)
+      choiceRunner(selected) 
+
+  } else if (willUserChoose === "self") {
+
+    let usersChoice = readline.question("OK! would you like [oatmeal], [sandwiches], or [smoothie] bowls? ");
+    choiceRunner(usersChoice);
+      
+  } else {
+    wantBreakfast = readline.question('Would you like to prepare a nice BREAKFAST to start your day, [y]es or [n]o? ');
+    // getAllRecipe();
+
+  }
+
+
+  askForGroceries = readline.question("Do you have all the groceries [y]es or [n]o? ");
+  if (askForGroceries === "y") {
+    console.log("Great let get started");
+    getRecipe();
+  } else {
+    console.log("Go do your groceries then come back :( ")
+  }
+  willUserChoose = readline.question("would you like to choose for your[self] or, be given a [random] option again? ");
+  usersChoice = readline.question("OK! would you like [oatmeal], [sandwiches], or [smoothie] bowls? ");
+  choiceRunner(usersChoice);
+
+  let willUserPrint = readline.question("would you like to print all the recipes [y]es or [n]o? ");
+  function getAllRecipe() {
+    console.clear();
+    ingredientsOvernightOat();
+    smoothieBowls();
+    fruitNutOatmeal();
+    eggBreakfastSandwich();
+    SmokedSalmonDillEggsSandwich();
+    GuacamoleBreakfastSandwich();
+  }
+  console.log(willUserPrint)
+  
+  if (willUserPrint === "y") {
+    
+    getAllRecipe()
+    
+  } else {
+
+    willUserChoose = readline.question("would you like to choose for your[self] or, be given a [random] option again? ");
+    usersChoice = readline.question("OK! would you like [oatmeal], [sandwiches], or [smoothie] bowls? ");
+    choiceRunner(usersChoice);
+
+  }
+
+
+
+  usersPantryCheatCheat = readline.question("OK! would you like pantry cheat cheat how to double recipes [y] or [n]? ");
+  console.log("Pantry Cheat Cheat How to Double Recipes")
+
+if (usersPantryCheatCheat === "y"){
+  const myForEach = ( arr, fn) => {
+    for ( let i = 0 ; i< arr.length ; i ++) {
+      fn([arr[i]])
+    }
+  }
+
```

Comment:
Here you can create one function where it takes in two arguments, one would be num and the other could be measuring container. Then you would only need to do one function and would be able to reuse it in all these instances.